### PR TITLE
Modularization of the main page

### PR DESCRIPTION
**Pull request by Juliana-Dimeglio**
Modularization of the main page 
- Creation of the widgets: coodinates_card, gesture_button and plugin_button
- Creation of the new class plugin_functionality
image

### CODE REVIEW

File: example/lib/widgets/coordinates_card.dart [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  @override
+  _CoordinatesCardState createState() => _CoordinatesCardState();
+}
+
+class _CoordinatesCardState extends State<CoordinatesCard> {
+  @override
+  Widget build(BuildContext context) {
+    return widget.hasData
+        ? Container(
+            padding: EdgeInsets.all(
```
Comment (FranciscoGaraventa) [inline example/lib/widgets/coordinates_card.dart]: This could be const

File: example/lib/widgets/coordinates_card.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                  )
+                : Text(
+                    widget.data.toString(),
+                    style: TextStyle(
+                      fontSize: NumericConstants.fontSizeLocationText,
+                    ),
+                  ),
+          )
+        : Container(
+            padding: EdgeInsets.all(
```
Comment (FranciscoGaraventa) [inline example/lib/widgets/coordinates_card.dart]: This could be const too

File: example/lib/widgets/coordinates_card.dart [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+              NumericConstants.paddingCoordinates,
+            ),
+            decoration: BoxDecoration(
+              border: Border.all(
+                color: Colors.black,
+              ),
+            ),
+            child: Text(
```
Comment (FranciscoGaraventa) [inline example/lib/widgets/coordinates_card.dart]: This Text could be const

File: example/lib/widgets/gesture_button.dart [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  const GestureButton({
+    Key? key,
+    required this.onTap,
+    required this.buttonText,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.all(
```
Comment (FranciscoGaraventa) [inline example/lib/widgets/gesture_button.dart]: This could be a const margin

File: example/lib/widgets/gesture_button.dart [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(
+          NumericConstants.borderRadiusGestureDetector,
+        ),
+        color: ColorsConstants.gestureDetectorColor,
+      ),
+      child: GestureDetector(
+        onTap: onTap,
+        child: Padding(
+          padding: EdgeInsets.symmetric(
```
Comment (FranciscoGaraventa) [inline example/lib/widgets/gesture_button.dart]: This padding could be const

File: example/lib/widgets/coordinates_card.dart (hunk 1 above)
Comment (Juliana-Dimeglio) [reply example/lib/widgets/coordinates_card.dart]: Fixed

File: example/lib/widgets/coordinates_card.dart (hunk 2 above)
Comment (Juliana-Dimeglio) [reply example/lib/widgets/coordinates_card.dart]: Fixed

File: example/lib/widgets/coordinates_card.dart (hunk 3 above)
Comment (Juliana-Dimeglio) [reply example/lib/widgets/coordinates_card.dart]: Fixed

File: example/lib/widgets/gesture_button.dart (hunk 4 above)
Comment (Juliana-Dimeglio) [reply example/lib/widgets/gesture_button.dart]: Fixed

File: example/lib/widgets/gesture_button.dart (hunk 5 above)
Comment (Juliana-Dimeglio) [reply example/lib/widgets/gesture_button.dart]: Fixed

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

**Pull request merged**